### PR TITLE
No SQL mode in web console

### PR DESCRIPTION
**Pull request by shuqi7**
- Added 'No SQL mode' in the web console where all data retrieved from the server is through sending GET requests to Druid APIs instead of running SQL queries, just like the legacy consoles
- No SQL mode is enabled when the broker for SQL query is down
- Client-side workaround to fix https://github.com/apache/incubator-druid/issues/7498 until Druid SQL returns servers which have no data

### CODE REVIEW

File: web-console/src/console-application.tsx [hunk 1]
```diff
@@ -62,28 +63,34 @@ export class ConsoleApplication extends React.Component<ConsoleApplicationProps,
...
+      /* tslint:enable:jsx-alignment */
+    });
+  }
+
+  private async setSqlMode() {
+    if (!(await ConsoleApplication.ensureSql())) {
+      ConsoleApplication.shownNotifications();
+      this.setState({
+        noSqlMode: true
+      });
```
Comment (vogievetsky) [inline web-console/src/console-application.tsx]: You can not just set noSqlMode like this, you need to also have a flag like `capabilitiesLoading` and you need to block the entire UI (show a loader) until this call returns. (I would wrap it in a QueryManager).
The first view will not know if it needs to use sql or not until this call completes

File: web-console/src/console-application.tsx [hunk 2]
```diff
@@ -62,28 +63,34 @@ export class ConsoleApplication extends React.Component<ConsoleApplicationProps,
...
+        It appears that the SQL endpoint is disabled. No SQL endpoint is
+        used and all requests rely on <a href={DRUID_DOCS_API}>native Druid query API</a>,
+        but you may still <a href={DRUID_DOCS_SQL}>enable the SQL endpoint</a>
+      </>
+      /* tslint:enable:jsx-alignment */
+    });
+  }
+
+  private async setSqlMode() {
+    if (!(await ConsoleApplication.ensureSql())) {
```
[the remark this answers is not in the document]
Comment (shuqi7) [reply web-console/src/console-application.tsx]: I think if we have `(response.status !== 405 || response.statusText !== 'Method Not Allowed')`, then SQL is still working right? And what is the case if for `broken`? I think we have a different Apptoast to indicate this state?

File: web-console/src/console-application.tsx [hunk 3]
```diff
@@ -45,45 +47,53 @@ export interface ConsoleApplicationProps extends React.Props<any> {
...
-  static async shownNotifications() {
-    await ConsoleApplication.ensureSql();
+  static shownNotifications(capabilities: string) {
+    let message: JSX.Element = <></>;
+    /* tslint:disable:jsx-alignment */
+    if (capabilities === 'working-without-sql') {
+      message = <>
+        It appears that the SQL endpoint is disabled. No SQL endpoint is
+        used and all requests rely on <a href={DRUID_DOCS_API}>native Druid query API</a>,
+        but you may still <a href={DRUID_DOCS_SQL}>enable the SQL endpoint</a>
```
Comment (vogievetsky) [inline web-console/src/console-application.tsx]: can this instead be ("No SQL endpoint" does not make sense):

```
It appears that the SQL endpoint is disabled. The console will fall back to <a href={DRUID_DOCS_API} target="_blank">native Druid APIs</a> and will be limited in functionality. Look at <a href={DRUID_DOCS_SQL} target="_blank">the SQL docs</a> to enable the SQL endpoint.
```

**Comment by vogievetsky** [comment]
Made a quick note on the wording of the notice and I think it is good otherwise. 👍 once my comment above is addressed.

File: web-console/src/views/servers-view.tsx [hunk 4]
```diff
@@ -100,14 +123,37 @@ export class ServersView extends React.Component<ServersViewProps, ServersViewSt
...
   componentDidMount(): void {
+    const { noSqlMode } = this.props;
     this.serverQueryManager = new QueryManager({
       processQuery: async (query: string) => {
-        const servers = await queryDruidSql({ query });
-
+        let servers: ServerQueryResultRow[];
+        if (!noSqlMode) {
+          servers = await queryDruidSql({ query });
+          if (servers.length === 0) {
```
Comment (vogievetsky) [inline web-console/src/views/servers-view.tsx]: could you add a comment here about why we are doing this, also link in the comment to https://github.com/apache/incubator-druid/issues/7498 and https://github.com/apache/incubator-druid/issues/7005

Comment (shuqi7) [reply web-console/src/views/servers-view.tsx]: Will implement another way of querying the servers in a separate PR

**Pull request merged**